### PR TITLE
Windows port and solved some issues

### PR DESCRIPTION
**Pull request by eulynx**
Added the Windows port and solved some issues. Try/catch must be improved and added on more places in the source code (one example is available).

### CODE REVIEW

**Comment by eulynx** [comment]
Please review the changes I have made

File: example.cpp [hunk 1]
```diff
@@ -1,64 +1,99 @@
+// ModbusPP.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
 //
 // Created by Fanzhe on 5/29/2017.
 //
 
-#include "modbus.h"
+#include "../modbuspp-master/modbus.h"
+#include "../modbuspp-master/modbus_exception.h"
```
Comment (Tectu) [inline example.cpp:9]: This seems like that should not have ended up in this commit. Those inclusion paths will not work for other people.

For the rest I'd be looking forward for this to get merged as I am planning to use this library on Windows soon.

Comment (fz-lyu) [reply example.cpp:9]: Yes In general the example file should be independent from the underlying definition.

**Comment by fz-lyu** [comment]
Most of the code got reimplemented in a more generalized way. Close the PR.

**Pull request closed without merging**